server/src/main/java/org/apache/druid/catalog/model/table/HttpTableDefn.java

```diff
@@ -0,0 +1,190 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.catalog.model.table;
+
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableMap;
+import org.apache.druid.catalog.model.CatalogUtils;
+import org.apache.druid.catalog.model.ModelProperties.StringListPropertyDefn;
+import org.apache.druid.catalog.model.ModelProperties.StringPropertyDefn;
+import org.apache.druid.catalog.model.ParameterizedDefn;
+import org.apache.druid.catalog.model.ResolvedTable;
+import org.apache.druid.catalog.model.table.ExternalTableDefn.FormattedExternalTableDefn;
+import org.apache.druid.data.input.InputSource;
+import org.apache.druid.data.input.impl.HttpInputSource;
+import org.apache.druid.java.util.common.IAE;
+import org.apache.druid.java.util.common.ISE;
+import org.apache.druid.java.util.common.StringUtils;
+import org.apache.druid.metadata.DefaultPasswordProvider;
+import org.apache.druid.metadata.EnvironmentVariablePasswordProvider;
+import org.apache.druid.utils.CollectionUtils;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Definition of an input table for an HTTP data source. Provides the same
+ * properties as the {@link HttpInputSource}, but as top-level properties
+ * that can be mapped to SQL function parameters. Property names are
+ * cleaned up for ease-of-use. The HTTP input source has multiple quirks,
+ * the conversion method smooths over those quirks for a simpler catalog
+ * experience. Provides a parameterized
+ * form where the user provides the partial URLs to use for a particular
+ * query.
+ */
+public class HttpTableDefn extends FormattedExternalTableDefn implements ParameterizedDefn
+{
+  public static final String TABLE_TYPE = HttpInputSource.TYPE_KEY;
+  public static final String URI_TEMPLATE_PROPERTY = "template";
```

Comment:
is `template` too generic? Maybe `urisTemplate`?

Reply:
This is one of those context questions. To you, reading this file amid zillions of others, the term "template" is too generic. To a user, focused on the HTTP input source, there is only one thing that could be templated: the URL. Since these values also appear in SQL, shorter names is a convenience for the user:

```SQL
... FROM (TABLE(http(template => 'http://foo.bar/{}.html')))
```
vs.
```SQL
... FROM (TABLE(http(httpInputSourceURLTemplate => 'http://foo.bar/{}.html')))
```

That said, changed it to `uriTemplate`. Three more letters to type and a shift key to hit...